src/server/project.ts

```diff
@@ -2212,6 +2212,40 @@ namespace ts.server {
         }
 
         updateReferences(refs: readonly ProjectReference[] | undefined) {
+            // @ts-ignore
+            if (process.versions.pnp) {
+                // With Plug'n'Play, dependencies that list peer dependencies
+                // are "virtualized": they are resolved to a unique (virtual)
+                // path that the underlying filesystem layer then resolve back
+                // to the original location.
+                //
+                // When a workspace depends on another workspace with peer
+                // dependencies, this other workspace will thus be resolved to
+                // a unique path that won't match what the initial project has
+                // listed in its `references` field, and TS thus won't leverage
+                // the reference at all.
+                //
+                // To avoid that, we compute here the virtualized paths for the
+                // user-provided references in our references by directly querying
+                // the PnP API. This way users don't have to know the virtual paths,
+                // but we still support them just fine even through references.
+
+                const pnpApi = require("pnpapi");
+                const basePath = this.getCurrentDirectory();
+
+                const getPnpPath = (path: string) => {

```

Comment:
@arcanis - This function is causing tsconfig references to fail to load right now in yarn `3.0.0`. pnpApi always seems to return the project root when `path` is absolute and lies within the `workspace:.` path. This is causing tsconfig.json files with referenced config json files within the workspace, to be invalid, only in tsserver somehow.

I tried and succeeded in fixing this issue locally by unplugging typescript and making a slight modification to this function -

```ts
const getPnpPath = (path: string) => {
    if (pathIsAbsolute(path)) return path; // intended compiled code - `if (ts.pathIsAbsolute(path)) return path;`
    try {
        const targetLocator = pnpApi.findPackageLocator(`${path}/`);
        return pnpApi.resolveToUnqualified(targetLocator.name, `${basePath}/`);
    } catch {
        // something went wrong with the resolution, try not to fail
        return path;
    }
};
```

please consider making this change or add a better solution for this issue :)

Thanks.